From kay.sievers@vrfy.org  Thu Jan 14 16:49:08 2010
From: Kay Sievers <kay.sievers@vrfy.org>
Date: Thu, 14 Jan 2010 22:49:39 +0100
Subject: Driver-Core: disable /sbin/hotplug by default
To: Greg KH <greg@kroah.com>
Message-ID: <1263505779.2074.3.camel@yio.site>


From: Kay Sievers <kay.sievers@vrfy.org>

No recent mainstream system uses the /sbin/hotplug fork-bomb any more.
Disable it by default to reflect how it is used these days.

Signed-off-by: Kay Sievers <kay.sievers@vrfy.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/base/Kconfig |   10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

--- a/drivers/base/Kconfig
+++ b/drivers/base/Kconfig
@@ -3,10 +3,18 @@ menu "Generic Driver Options"
 config UEVENT_HELPER_PATH
 	string "path to uevent helper"
 	depends on HOTPLUG
-	default "/sbin/hotplug"
+	default ""
 	help
 	  Path to uevent helper program forked by the kernel for
 	  every uevent.
+	  Before the switch to the netlink-based uevent source, this was
+	  used to hook hotplug scripts into kernel device events. It
+	  usually pointed to a shell script at /sbin/hotplug.
+	  This should not be used today, because usual systems create
+	  many events at bootup or device discovery in a very short time
+	  frame. One forked process per event can create so many processes
+	  that it creates a high system load, or on smaller systems
+	  it is known to create out-of-memory situations during bootup.
 
 config DEVTMPFS
 	bool "Maintain a devtmpfs filesystem to mount at /dev"
